examples/drivers/windows/otLwf/settings.c

```diff
@@ -39,80 +39,497 @@ void otPlatSettingsInit(otInstance *otCtx)
 {
     NT_ASSERT(otCtx);
     PMS_FILTER pFilter = otCtxToFilter(otCtx);
-    UNREFERENCED_PARAMETER(pFilter);
+
+    DECLARE_CONST_UNICODE_STRING(SubKeyName, L"OpenThread");
+
+    OBJECT_ATTRIBUTES attributes;
+    ULONG disposition;
+
+    InitializeObjectAttributes(
+        &attributes,
+        (PUNICODE_STRING)&SubKeyName,
+        OBJ_CASE_INSENSITIVE | OBJ_KERNEL_HANDLE,
+        pFilter->InterfaceRegKey,
+        NULL);
+
+    // Create/Open the 'OpenThread' sub key
+    NTSTATUS status =
+        ZwCreateKey(
+            &pFilter->otSettingsRegKey,
+            KEY_ALL_ACCESS,
+            &attributes,
+            0,
+            NULL,
+            REG_OPTION_NON_VOLATILE,
+            &disposition);
+
+    NT_ASSERT(NT_SUCCESS(status));
+    if (!NT_SUCCESS(status))
+    {
+        LogError(DRIVER_DEFAULT, "ZwCreateKey for 'OpenThread' key failed, %!STATUS!", status);
+    }
+}
+
+uint16_t FilterCountSettings(_In_ PMS_FILTER pFilter, uint16_t aKey)
+{
+    HANDLE regKey = NULL;
+    OBJECT_ATTRIBUTES attributes;
+    DECLARE_UNICODE_STRING_SIZE(Name, 8);
+    UCHAR InfoBuffer[128] = {0};
+    PKEY_FULL_INFORMATION pInfo = (PKEY_FULL_INFORMATION)InfoBuffer;
+    ULONG InfoLength = sizeof(InfoBuffer);
+
+    // Convert 'aKey' to a string
+    RtlIntegerToUnicodeString((ULONG)aKey, 16, &Name);
+
+    InitializeObjectAttributes(
+        &attributes,
+        &Name,
+        OBJ_CASE_INSENSITIVE | OBJ_KERNEL_HANDLE,
+        pFilter->otSettingsRegKey,
+        NULL);
+
+    // Open the registry key
+    NTSTATUS status =
+        ZwOpenKey(
+            &regKey,
+            KEY_ALL_ACCESS,
+            &attributes);
+
+    if (!NT_SUCCESS(status))
+    {
+        // Key doesn't exist, return a count of 0
+        goto error;
+    }
+
+    // Query the key info from the registry
+    status =
+        ZwQueryKey(
+            regKey,
+            KeyValueFullInformation,
+            pInfo,
+            InfoLength,
+            &InfoLength);
+
+    if (!NT_SUCCESS(status))
+    {
+        LogError(DRIVER_DEFAULT, "ZwQueryKey for %S value failed, %!STATUS!", Name.Buffer, status);
```

Comment:
I explicitly initialize the `InfoBuffer` buffer to `{0}` and since `pInfo` points to that buffer, in the case that `ZwQueryKey` fails, `pInfo->Values` should still be `0`.